src/main/java/com/epam/aidial/core/storage/BlobStorage.java

```diff
@@ -0,0 +1,199 @@
+     * This method must be called if something was wrong during upload to clean up uploaded parts/chunks
+     */
+    public void abortMultipartUpload(MultipartUpload multipart) {
+        blobStore.abortMultipartUpload(multipart);
+    }
+
+    /**
+     * Upload file in a single request
+     *
+     * @param fileName    name of the file, for example: data.csv
+     * @param path        absolute path according to the bucket, for example: Users/user1/files/input data
+     * @param contentType MIME type of the content, for example: text/csv
+     * @param data        whole content data
+     */
+    public void store(String fileName, String path, String contentType, Buffer data) {
+        String filePath = BlobStorageUtil.buildFilePath(fileName, path);
+        Blob blob = blobStore.blobBuilder(filePath)
+                .payload(new BufferPayload(data))
+                .contentLength(data.length())
+                .contentType(contentType)
+                .build();
+
+        String etag = blobStore.putBlob(bucketName, blob);
+        log.info("Stored etag: " + etag);
+    }
+
+    /**
+     * Load file content from blob store
+     *
+     * @param filePath absolute file path, for example: Users/user1/files/inputs/data.csv
+     * @return Blob instance if file was found, null - otherwise
+     */
+    public Blob load(String filePath) {
+        return blobStore.getBlob(bucketName, filePath);
+    }
+
+    /**
+     * Delete file content from blob store
+     *
+     * @param filePath absolute file path, for example: Users/user1/files/inputs/data.csv
+     */
+    public void delete(String filePath) {
+        blobStore.removeBlob(bucketName, filePath);
+    }
+
+    /**
+     * List all files/folder metadata for a given path
+     *
+     * @param path absolute path for a folder, for example: Users/user1/files
+     */
+    public List<FileMetadataBase> listMetadata(String path) {
+        List<FileMetadataBase> metadata = new ArrayList<>();
+        ListContainerOptions options = buildListContainerOptions(BlobStorageUtil.normalizePathForQuery(path));
+        PageSet<? extends StorageMetadata> list = blobStore.list(bucketName, options);
+        list.forEach(meta -> {
+            StorageType objectType = meta.getType();
+            switch (objectType) {
+                case BLOB -> metadata.add(buildFileMetadata(meta));
+                case FOLDER, RELATIVE_PATH -> metadata.add(buildFolderMetadata(meta));
+                default -> throw new IllegalArgumentException("Can't list container");
+            }
+        });
+
+        return metadata;
+    }
+
+    public void close() {
+        storeContext.close();
+    }
+
+    private static ListContainerOptions buildListContainerOptions(String prefix) {
+        return new ListContainerOptions()
+                .prefix(prefix)
+                .delimiter(BlobStorageUtil.PATH_SEPARATOR);
+    }
+
+    private static FileMetadata buildFileMetadata(StorageMetadata metadata) {
```

Comment:
These two methods do the same work. Wouldn't it be possible to merge them into one?

Reply:
fixed